updatecli/scripts/check-jdk.sh

```diff
@@ -30,10 +30,10 @@ function get_jdk_download_url() {
       echo "https://github.com/adoptium/temurin19-binaries/releases/download/jdk-${jdk_version}/OpenJDK19U-jdk_${platform}_hotspot_${jdk_version//+/_}";
       return 0;;
     21*)
-      # JDK version (21+35-ea-beta)
-      ##    https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35-ea-beta/OpenJDK21U-jdk_aarch64_linux_hotspot_ea_21-0-35.tar.gz
+      # JDK version (21+35)
+      ##    https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35/OpenJDK21U-jdk_aarch64_linux_hotspot_21_35.tar.gz
       urlEncodedJDKVersion="${jdk_version//+/%2B}"
-      echo "https://github.com/adoptium/temurin21-binaries/releases/download/jdk-${urlEncodedJDKVersion}/OpenJDK21U-jdk_${platform}_hotspot_ea_21-0-$(echo ${jdk_version} | cut -d '+' -f 2 | cut -d '-' -f 1)"
+      echo "https://github.com/adoptium/temurin21-binaries/releases/download/jdk-${urlEncodedJDKVersion}/OpenJDK21U-jdk_${platform}_hotspot_${jdk_version//+/_}"
```

Comment:
```suggestion
      # TODO: the "-debugimage" might not stay in these URLs, to be removed when it's the case
      echo "https://github.com/adoptium/temurin21-binaries/releases/download/jdk-${underscoredJDKVersion}/OpenJDK21U-debugimage_${platform}_hotspot_${underscoredJDKVersion}"
```

Reply:
Why do we need to track those debug image ?

Reply:
Nevermind, yesterday they were the only ones available, didn't check the link again today.

Reply:
Edited my suggestion 🙂

Reply:
we don't want to use the debug images. See my comment in https://github.com/jenkins-infra/jenkins-infra/pull/3113#discussion_r1358029572

Reply:
debug image is a different image and not what we want